pkg/components/linkerd/component.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linkerd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/linkerd/linkerd2/pkg/tls"
+	"helm.sh/helm/v3/pkg/chartutil"
+
+	internaltemplate "github.com/kinvolk/lokomotive/internal/template"
+	"github.com/kinvolk/lokomotive/pkg/components"
+	"github.com/kinvolk/lokomotive/pkg/components/util"
+	"github.com/kinvolk/lokomotive/pkg/k8sutil"
+)
+
+const (
+	name = "experimental-linkerd"
+)
+
+// nolint:gochecknoinits
+func init() {
+	components.Register(name, newComponent())
+}
+
+type component struct {
+	ControllerReplicas int `hcl:"controller_replicas,optional"`
```

Comment:
Hm, it seems changing this parameter does not do anything. Also, is 0 or -1 number of replicas a valid value?

Reply:
> Hm, it seems changing this parameter does not do anything.

What do you mean by does not do anything?

```console
$ lokoctl component render-manifest experimental-linkerd | grep replicas
  resources: ["daemonsets", "deployments", "replicasets", "statefulsets"]
  replicas: 3
  resources: ["daemonsets", "deployments", "replicasets", "statefulsets"]
  replicas: 3
  replicas: 3
  replicas: 1
  resources: ["replicasets"]
  replicas: 3
  replicas: 1
  replicas: 1
  resources: ["replicasets"]
  replicas: 3
  replicas: 3
  resources: ["deployments", "replicasets", "daemonsets", "statefulsets"]
```

```console
$ lokoctl component render-manifest experimental-linkerd | grep replicas
  resources: ["replicasets"]
  resources: ["daemonsets", "deployments", "replicasets", "statefulsets"]
  replicas: 2
  replicas: 1
  replicas: 1
  resources: ["deployments", "replicasets", "daemonsets", "statefulsets"]
  replicas: 1
  replicas: 2
  resources: ["daemonsets", "deployments", "replicasets", "statefulsets"]
  replicas: 2
  replicas: 2
  replicas: 2
  resources: ["replicasets"]
  replicas: 2
```

> Also, is 0 or -1 number of replicas a valid value?

Added a validation.

Reply:
>What do you mean by does not do anything?

Hmm, odd. I've tried that and I couldn't spot the change :thinking: Thanks for checking then!